### DISCUSSION TITLE
skywalking php-agent mysqli metrics

### DISCUSSION THREAD

**Discussion by thecalmund** [apache/skywalking#9943] [Q&A] [answered]
Hello,

i have running the skywalking php extension on my webserver machine with php7.4 and im using the php-mysqli extension but Skywalking doesn't get any mysql metrics in a request trace. Do I need to enable something in the agent config section in my php.ini? The docs aren't clear about that.

Thank you for helping

**Comment by wu-sheng** [comment, marked as answer]
I can't see any plugin relative to mysqli on the PHP document. Why do you think it is supported?

**Reply by thecalmund** [reply]
Like you can see in my picture there is a section with the supported php plugins and there is also the php-mysqli module listed:
![image](https://user-images.githubusercontent.com/117904609/201064898-4bf885b0-5642-48de-8382-58e55571eab5.png)

**Reply by wu-sheng** [reply]
This is doc for next release. Are you building from GitHub?

**Reply by thecalmund** [reply]
Hi, now I builded it from GitHub and now I don't get metrics overall. The error in the log is the following:
```
2022-11-10T11:13:10.180573Z DEBUG skywalking_agent::plugin::plugin_redis: Get redis peer addr="XXX.XXX.XXX:6379"
2022-11-10T11:13:10.185285Z DEBUG skywalking_agent::plugin::plugin_redis: call redis method handle=618 function_name="get"
2022-11-10T11:13:10.185308Z DEBUG skywalking_agent::plugin::plugin_redis: call redis command handle=618 function_name="get" key="identData:6__0_0" op="read"
2022-11-10T11:13:10.223983Z ERROR shutdown: skywalking_agent::channel: channel send failed err=Permission denied (os error 13)
2022-11-10T11:13:10.224086Z DEBUG skywalking_agent::plugin::plugin_redis: call Redis free
```

Do I have to deactivate the redis plugin in the agent config section for example? I only need the mysqli metrics.

**Reply by wu-sheng** [reply]
It is a tracing plugin, not metric at all.

Read things more carefully. 

@jmjoy FYI

**Reply by jmjoy** [reply]
The PHP plugin only support trace now, metric and log maybe will support in future.

The document needs to be clearer, thanks your feedback.

**Reply by jmjoy** [reply]
> Hi, now I builded it from GitHub and now I don't get metrics overall. The error in the log is the following:
> 
> ```
> 2022-11-10T11:13:10.180573Z DEBUG skywalking_agent::plugin::plugin_redis: Get redis peer addr="XXX.XXX.XXX:6379"
> 2022-11-10T11:13:10.185285Z DEBUG skywalking_agent::plugin::plugin_redis: call redis method handle=618 function_name="get"
> 2022-11-10T11:13:10.185308Z DEBUG skywalking_agent::plugin::plugin_redis: call redis command handle=618 function_name="get" key="identData:6__0_0" op="read"
> 2022-11-10T11:13:10.223983Z ERROR shutdown: skywalking_agent::channel: channel send failed err=Permission denied (os error 13)
> 2022-11-10T11:13:10.224086Z DEBUG skywalking_agent::plugin::plugin_redis: call Redis free
> ```
> 
> Do I have to deactivate the redis plugin in the agent config section for example? I only need the mysqli metrics.

The ERROR `Permission denied` looks like there's a problem, would you create issue about it?

**Reply by jmjoy** [reply]
@thecalmund Please try this PR https://github.com/apache/skywalking-php/pull/33 , check whether the problem of Permission denied has been solved.

**Reply by thecalmund** [reply]
Yes, it is working. Thanks 👍
